lib/matplotlib/figure.py

```diff
@@ -2109,6 +2109,10 @@ def __init__(self,
         # a proxy property), but that actually need to be on the figure for
         # pickling.
         self._canvas_callbacks = cbook.CallbackRegistry()
+        self._button_pick_id = self._canvas_callbacks.connect(
```

Comment:
I'm not particularly up on how the callback system works, but this seems intrusive in `figure.py`.

Reply:
The point of https://github.com/matplotlib/matplotlib/pull/16220 was to move callback handling from the canvas to the figure (for reasons explained in that PR).